### ISSUE TITLE
Check for a valid trace for ALL tests

### ISSUE THREAD

**Issue by plbossart** [thesofproject/sof-test#297]
We need to double-check that the trace is functional and that sof-logger reports at least the firmware details and a dai trigger.

If there is no trace then there's no point in checking results and debugging further.

cc:
- #1075
- https://github.com/thesofproject/sof/pull/7955
- #1075
- https://github.com/thesofproject/sof-test/pull/1084

+ many others, follow the links.

**Comment by marc-hb** [comment]
First part submitted in #373, please help review.

**Comment by marc-hb** [comment]
From @xiulipan :
> it that behavior right? Do we need to fail the case if we could not get the logger? As we have standalone test case for logger.

**Comment by marc-hb** [comment]
> Do we need to fail the case if we could not get the logger? 

I think what you're asking is: "is the logger a low priority feature"? I read the description of this request as a pretty clear "no" but I'd like to hear more from others.

> As we have standalone test case for logger.

It's risky to rely on a single test, if it's broken then the feature can be broken and no one will notice.  Even if the test is not broken, the logger could be broken only in some scenarios not tested by the logger test. I think the links above are some real-world examples of either.

A test is never meant to test only one thing, the more extras each test can test (in no extra time) and the better. Unexpected failures mean extra bugs are found which is great.

These questions sounds strange to me at a time when the logger seems to be working fine, I mean  all tests in #373 pass right now. I mean it sounds very strange to worry about failures we don't even have right now (this reminds me of recent #372) It almost sounds like like a desire to find FEWER bugs / and have more "green failures"?

**Comment by xiulipan** [comment]
@marc-hb What I want to make clarify about is the **criterion** for our test case pass/fail.
Should we make aplay test fail if a debug feature is not working?

> A test is never meant to test only one thing, the more extras each test can test (in no extra time) and the better. 

I do not think we are adding test cases in this way. Each test case in this repo is focusing in a single feature. This will help us to narrow down the issue.

**Comment by marc-hb** [comment]
> I do not think we are adding test cases in this way. Each test case in this repo is focusing in a single feature. This will help us to narrow down the issue.

This is :
- neither possible because sof-test are not unit tests. They all run from user space so each test involves many lines of code
- nor desirable unless you want to find fewer bugs - as it happened in the links above and elsewhere.

**Comment by marc-hb** [comment]
>  I mean all tests in #373 pass right now.

They didn't pass, they only appear to pass because I misused `exit_failure=2` in #373 where 2 means "Not Applicable" and Not Applicable is green. I didn't know 2 was a special value.

> I mean it sounds very strange to worry about [logger] failures we don't even have right now

Now we have one: pretty much all tests involving the logger fail to kill -INT it at the end, see new bug https://github.com/thesofproject/sof/issues/3433 [`BUG] sof-logger resisted pkill -INT, using -KILL`

**Comment by marc-hb** [comment]
multipipeline tests returning an empty trace: https://github.com/thesofproject/sof-test/pull/380#issuecomment-694626882

**Comment by marc-hb** [comment]
`multiple-pipeline-capture.sh` and `multiple-pipeline-playback.sh` have a consistently empty trace on about half the platforms: https://sof-ci.01.org/softestpr/PR383/build194/devicetest/

```
2020-09-18 05:13:41 UTC [REMOTE_INFO] nlines=1 /home/ubuntu/sof-test/logs/multiple-pipeline-playback/2020-09-18-05:13:29-16938/slogger.txt
2020-09-18 05:13:41 UTC [REMOTE_ERROR] Empty logger trace
2020-09-18 05:13:41 UTC [REMOTE_INFO] Test Result: FAIL!
```

@aiChaoSONG  does this seem related to your #359 PR?

**Comment by aiChaoSONG** [comment]
@marc-hb Yes, they are related. without #359, these two case are actually testing nothing, and always pass. It is the sof-logger issue  who makes the multiple-pipeline-playback/capture failed after your  https://github.com/thesofproject/sof-test/pull/373 patch

**Comment by marc-hb** [comment]
I realized nothing looks for any error in the SOF logs, they go completely unnoticed. For instance we had ` src/lib/wait.c:46   10401 ERROR ewt` for ages yet that never caused any test to fail. Maybe we need a new sof-test issue for this other validation gap.

On a more positive note, I've been submitting a number of logger-related fixes that feel like progress. Please help review the ones not merged yet:

sof-test
hijack.sh: fail if the logger trace is empty #662
hijack.sh: make late logger config changes to nudge stuck DMA traces #663
check-sof-logger: rewrite most of it so it can actually find bugs #666

sof
trace.c: don't split mbox events over the buffer's end and lose them  https://github.com/thesofproject/sof/pull/4083
logger: print a separator when the timestamp goes back in time https://github.com/thesofproject/sof/pull/4084
logger: warn user about wrapped logs and skipped garbage https://github.com/thesofproject/sof/pull/4085
logger: fix column and header aligments https://github.com/thesofproject/sof/pull/4086

  \+ some other minor ones not worth mentioning here and a couple last more coming, I'll keep updating that comment.

**Comment by marc-hb** [comment]
After finally testing this a bit (#666) and performing some overdue "data-mining" in test results it's very clear the DMA trace has been very unreliable: `Empty DMA trace` #4333. On some platforms much more than others.

**Comment by marc-hb** [comment]
More DMA unreliability:
- thesofproject/sof/pull/4844
- thesofproject/sof/issues/5120

**Comment by mengdonglin** [comment]
Low priority to P2 as SOF is going to use native Zephyr log implementation.

**Comment by marc-hb** [comment]
I don't understand, Zephyr bugs can be solved without any logs?

**Comment by marc-hb** [comment]
- https://github.com/thesofproject/sof-test/pull/897

... should finish this but it's currently blocking, triggering some failures.

- https://github.com/thesofproject/sof/issues/6039
- https://github.com/thesofproject/sof/issues/6042

**Comment by mengdonglin** [comment]
Blocked by mtrace enabling atm.

**Comment by marc-hb** [comment]
This has been finally fixed by @kv2019i:
- https://github.com/thesofproject/sof-test/pull/1095

And it does catch mtrace issues like https://github.com/thesofproject/sof-test/issues/1077 and others.

The test works, closing.
